### PR TITLE
Domains: Remove "Create site" button from domain-only sites pages

### PR DESCRIPTION
**Pull request by rafaelgallani**
## Proposed Changes
See here for more information: p1689230922152949/1689225408.499109-slack-C0BNMNMNG.
Tl;dr: In order to proceed with the new domain transfer/domain-only sites changes, this PR temporarily removes the "Create site" button from all domain-only site pages. 

## Testing Instructions
- Analyze the code statically;
- Ensure all unit tests are passing;
- Confirm that all references to the "Create site" button were removed:
  - Try purchasing a domain for a domain-only site and ensure that the button isn't displayed on the post-checkout page;
  - Go to a domain-only management page (`/domains/manage/:site`) and ensure that the button isn't displayed;
  - Ensure that on the "All domains" page, there's no upsell card with the "Create site" button;

#### Before
image

#### After
image

#### Before
image

#### After
image

#### Before
image

#### After
image

#### Before
image

#### After
image

## Pre-merge Checklist

- [X] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [X] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [X] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Comment by worldomonation** [comment]
I was able to verify this for one scenario:
- user signed up as part of the domain transfer flow = meaning no site associated with the user.
- user cancelled all domain transfers, leaving them with no site and no domains.

Compare what's on staging vs the same page loaded on the calypso.live of this branch. 
The "Create site" button is gone.

https://github.com/Automattic/wp-calypso/assets/6549265/85dddea6-a592-4cc2-82cc-88b626cf7163
